src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -871,48 +877,95 @@ void makeNotNullMembersMaybeNull()
                     }
                 }
 
+                return;
+
                 ImmutableArray<Symbol> getMembersNeedingDefaultInitialState()
                 {
                     if (_hasInitialState)
                     {
                         return ImmutableArray<Symbol>.Empty;
                     }
 
+                    bool includeRequiredMembers = true;
+                    bool includeBaseRequiredMembers = true;
+                    bool hasThisConstructorInitializer = false;
+
+                    if (method is SourceMemberMethodSymbol { SyntaxNode: ConstructorDeclarationSyntax { Initializer: { RawKind: var initializerKind } } })
+                    {
+                        var baseOrThisInitializer = (_baseOrThisInitializer ?? GetConstructorThisOrBaseSymbol(this.methodMainNode));
+                        // If there's an error in the base or this initializer, presume that we should set all required members to default.
+                        includeBaseRequiredMembers = baseOrThisInitializer?.ShouldCheckRequiredMembers() ?? true;
+                        if (initializerKind == (int)SyntaxKind.ThisConstructorInitializer)
```

Comment:
I don't feel super strongly about this, but we should avoid doing syntax checks after initial binding. Can we add the required information to bound nodes?

Reply:
> I don't feel super strongly about this, but we should avoid doing syntax checks after initial binding

Note that we were doing this syntax check anyway: it's part of `HasThisConstructorInitializer`. Given that, I don't want to change the logic any further.